src/handlers/motions/motionRewardClaimed/helpers.ts

```diff
@@ -0,0 +1,50 @@
+import { mutate } from '~amplifyClient';
+import { StakerReward } from '~types';
+import { getColonyFromDB } from '~utils';
+
+export const updateColonyUnclaimedStakes = async (
+  colonyAddress: string,
+  motionDatabaseId: string,
+  updatedStakerRewards: StakerReward[],
+): Promise<void> => {
+  const colony = await getColonyFromDB(colonyAddress);
+  if (colony) {
+    const { motionsWithUnclaimedStakes } = colony;
+    const motionWithUnclaimedStakes = motionsWithUnclaimedStakes?.find(
+      ({ motionId }) => motionDatabaseId === motionId,
+    );
+
+    if (motionWithUnclaimedStakes) {
+      const unclaimedRewards = updatedStakerRewards.filter(
+        ({ isClaimed }) => !isClaimed,
+      );
+
+      if (unclaimedRewards.length) {
```

Comment:
When a reward is claimed, we check to see if there are any unclaimed rewards remaining for the motion. 

If not, we remove the motion from the `motionsWithUnclaimedStakes` array. This is to make user-specific filtering in the client quicker.

If so, we update the motion with the current set of unclaimed rewards.

Reply:
Would be nice to have this comment in-code